### PR TITLE
2021 SIG Scalability annual report

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/community/issues/5509

/assign @mrbobbytables

### CODE REVIEW

**Comment by wojtek-t** [comment]
@mrbobbytables - I think this is ready for the first pass of review (to check if it is going in the right direction).

File: sig-scalability/2020-annual-report.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+**What areas and/or subprojects does the group need the most help with?**
+
+Each subproject could benefit from additional hands.
+However, the `Scalability Test Frameworks` and `Scalability and Performance tests and validation`
+are the ones where we can grow contributors - the other require both very deep and wide
+understanding of Kubernetes before making reasonable contributions.
+
+**What's the average open days of a PR and Issue in your group? / what metrics does your group care about and/or measure?**
+
+TODO: How to easily answer it post-factum?
```
Comment (mrbobbytables) [inline sig-scalability/2020-annual-report.md:109]: [Devstats](https://k8s.devstats.cncf.io/d/12/dashboards?orgId=1&refresh=15m) is where we pull a lot of metrics, but there are some current limitations that make it difficult to tease out information for SIGs that work out of k/k or other large shared repos =/

This is something ContribEx is working on with the CNCF.

Some of the devstats [PR](https://k8s.devstats.cncf.io/d/12/dashboards?viewPanel=9&orgId=1&refresh=15m) and [Issue](https://k8s.devstats.cncf.io/d/12/dashboards?viewPanel=8&orgId=1&refresh=15m) velocity boards might be useful from a general health standpoint, but if they're not - don't feel the need to include them.

If you have some metrics you'd like to see, something that could potentially help you manage your workload or SIG better, that would also be a great answer for this one.

File: sig-scalability/2020-annual-report.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+## Operational
+
+**How are you doing with operational tasks in sig-governance.md?**
+
+- **Is your README accurate? have a CONTRIBUTING.md file?**
+
+The README is accurate an up-to-date.
+We don't have CONTRIBUTING.md and it's in our short-term plans to update
+the [developer documentation].
```
Comment (mrbobbytables) [inline sig-scalability/2020-annual-report.md:11]: Thank you 🙏 
Just an FYI - If you don't have the bandwidth to update the developer docs, [calling out the things things that are out-dated and/or should be added](https://github.com/kubernetes/community/issues/5236) would be a great start.

**Review by mrbobbytables — COMMENTED** [review]
Thanks @wojtek-t things are looking good so far :) I left a few comments, but they're mostly FYIs.

**Comment by wojtek-t** [comment]
@mrbobbytables - PTAL when you will have a chance

File: sig-scalability/2020-annual-report.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+## Membership
+
+**Are all listed SIG leaders (chairs, tech leads, and subproject owners) active?**
+
+Yes
+
+**How do you measure membership? By mailing list members, OWNERs, or something else?**
+
+Active participants in SIG meetings and/or sending/reviewing PRs.
+TBD
```
Comment (mrbobbytables) [inline sig-scalability/2020-annual-report.md:50]:
```suggestion
```
This can probably be removed 👍

File: sig-scalability/2020-annual-report.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+Each subproject could benefit from additional hands.
+However, the `Scalability Test Frameworks` and `Scalability and Performance tests and validation`
+are the ones where we can grow contributors - the other require both very deep and wide
+understanding of Kubernetes before making reasonable contributions.
+
+**What's the average open days of a PR and Issue in your group? / what metrics does your group care about and/or measure?**
+
+We weren't measuring it unfortunately.
+What what be useful is out-of-the-box support for that across repositories
+(in particular perf-tests repo is very interesting for us).
```
Comment (mrbobbytables) [inline sig-scalability/2020-annual-report.md:116]: Just an FYI, we will soon be able to monitor this for individual repos. Devstats is being updated to include per repo and per sig (by label) filters for every dashboard.

File: sig-scalability/2020-annual-report.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+**How do you measure membership? By mailing list members, OWNERs, or something else?**
+
+Active participants in SIG meetings and/or sending/reviewing PRs.
+TBD
+
+**How does the group measure reviewer and approver bandwidth?
+Do you need help in any area now? What are you doing about it?**
+
+We're not doing a very good here. It happens that some PRs are waiting for weeks.
```
Comment (mrbobbytables) [inline sig-scalability/2020-annual-report.md:55]: One thing that might help would be a general review of the current [subprojects + associated OWNERS files](https://github.com/kubernetes/community/tree/master/sig-scalability) to make sure they're up to date + potentially promote more active folk into the reviewers.

**Review by mrbobbytables — CHANGES_REQUESTED** [review]
Content wise looks good. (one small nit)

One thing regarding the file name, it should be `annual-report-2021.md` There was some confusion around which year and if it should be on the front/back, which has since been cleared up (I made the same mistake with the contribex report).

I'll fire off a notice to the scalability list+steering for final commentary period and we can merge EoD 3/26.

File: sig-scalability/2020-annual-report.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+Are there any springing up or being retired? Are OWNERS files up to date
+in these areas?**
+
+There isn't any formal process. Updates are given during regular SIG meetings.
+
+**When was your last monthly community-wide update? (provide link to deck and/or recording)**
+
+January'20: [slide deck]
+
+[slide deck]:https://docs.google.com/presentation/d/1M81X3_SWwHrJaWdJsecJReOVNxjjWSBFBkhPKEV35hs/edit#slide=id.g401c104a3c_0_0
```
Comment (liggitt) [inline sig-scalability/2020-annual-report.md:38]: share the deck with kubernetes-dev?

File: sig-scalability/2020-annual-report.md (hunk 5 above)
Comment (liggitt) [inline sig-scalability/2020-annual-report.md:55]: typo?

```suggestion
We're not doing a very good job here. It happens that some PRs are waiting for weeks.
```

File: sig-scalability/2020-annual-report.md (hunk 6 above)
Comment (wojtek-t) [reply sig-scalability/2020-annual-report.md]: done

File: sig-scalability/2020-annual-report.md (hunk 3 above)
Comment (wojtek-t) [reply sig-scalability/2020-annual-report.md]: done

File: sig-scalability/2020-annual-report.md (hunk 5 above)
Comment (wojtek-t) [reply sig-scalability/2020-annual-report.md]: done

Comment (wojtek-t) [reply sig-scalability/2020-annual-report.md]: Subprojects definitions are up-to-date (I read that when creating this report).

That said - the adding more reviewers (especially for the perf-tests repo) is a good suggestion. I added it to the agenda of the next SIG meeting (next week).

**Review by wojtek-t — COMMENTED** [review]
Thanks for comments Bob and Jordan - PTAL.

File: sig-scalability/annual-report-2021.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+January'20: [slide deck]
+
+[slide deck]:https://docs.google.com/presentation/d/1M81X3_SWwHrJaWdJsecJReOVNxjjWSBFBkhPKEV35hs/edit
+
+
+## Membership
+
+**Are all listed SIG leaders (chairs, tech leads, and subproject owners) active?**
+
+Yes
```
Comment (nikhita) [inline sig-scalability/annual-report-2021.md:45]: How do you ensure that OWNERS files are upto date and all owners are active?

File: sig-scalability/annual-report-2021.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+Do you need help in any area now? What are you doing about it?**
+
+We're not doing a very good job here. It happens that some PRs are waiting for weeks.
+
+**Is there a healthy onboarding and growth path for contributors in your SIG?
+What are some activities that the group does to encourage this?
+What programs are you participating in to grow contributors throughout the contributor ladder?**
+
+We maintain a list of `help-wanted` issues and recommend particular ones for people
+interested in conctributing.
```
Comment (nikhita) [inline sig-scalability/annual-report-2021.md:61]:
```suggestion
interested in contributing.
```

Comment (wojtek-t) [reply sig-scalability/annual-report-2021.md]: done

**Comment by wojtek-t** [comment]
@mrbobbytables - PTAL when you will have a chance

**Comment by mrbobbytables** [comment]
/lgtm

Thanks @wojtek-t :)

**Pull request merged**